compiler/rustc_data_structures/src/sync.rs

```diff
@@ -52,6 +52,8 @@ pub use vec::{AppendOnlyIndexVec, AppendOnlyVec};
 
 mod vec;
 
+pub type MetadataRef = OwnedSlice;
```

Comment:
This type alias doesn't really belong here anymore when it's not cfged anymore.

Reply:
Do you have an idea of where to put it? (do we even need it now?)

Reply:
Deleting it is probably the best idea